sycl/include/CL/sycl/accessor.hpp

```diff
@@ -727,7 +727,10 @@ class accessor :
 
 public:
   // Default constructor for objects later initialized with __init member.
-  accessor() : impl({}) {}
+  accessor()
```

Comment:
I believe this c'tor should be private. @Fznamznon Could you comment?

Reply:
Even if so, it's not related to removing default `range` constructor and should committed separately.

Reply:
It's just a comment not "requested change".

Reply:
No, it can be defined only for device, but it can't be private, since we use Sema routines for OpenCL kernel generation.

Reply:
Is it different from using __init method(which is private)?

Reply:
Yes, we construct the `__init` call manually so there is no Sema diagnostics, but we construct accessor objects using Sema routines which diagnose incorrect member usages.